core-bundle/src/Menu/FrontendMenuBuilder.php

```diff
@@ -76,81 +76,118 @@ public function getMenu(ItemInterface $root, int $pid, array $options = []): Ite
             'isSitemap' => false,
         ], $options);
 
-        $pages = $this->getPages($pid, $options);
+        $item = null;
 
-        $request = $this->requestStack->getCurrentRequest();
-        /** @var PageModel|int|null $currentPage */
-        $currentPage = $request->attributes->get('pageModel');
+        if (0 !== $pid && null !== ($currentPage = $this->pageModelAdapter->findWithDetails($pid))) {
```

Comment:
Here you query the database for every page again, even though we already have the page model down there in line 99.

Reply:
It should already be in the registry then.

Reply:
As expected, there is no difference in re using the page object - as the the page is already in the registry.